AlphaWallet/Common/Services/ContentLoader.swift

```diff
@@ -0,0 +1,233 @@
+//
+//  ImageLoader.swift
+//  AlphaWallet
+//
+//  Created by Vladyslav Shepitko on 01.02.2023.
+//
+
+import UIKit
+import Combine
+import AVKit
+import Kingfisher
+import AlphaWalletFoundation
+
+protocol ContentCacheStorage: AnyObject {
+    func value(for key: String) throws -> Data?
+    func set(data: Data, for key: String) throws
+}
+
+final class KingfisherImageCacheStorage: ContentCacheStorage {
+    private let diskStorage = ImageCache(name: "aw-cache-store").diskStorage
+
+    func value(for key: String) throws -> Data? {
+        try diskStorage.value(forKey: key)
+    }
+
+    func set(data: Data, for key: String) throws {
+        try diskStorage.store(value: data, forKey: key)
+    }
+}
+
+protocol ContentLoaderResponseInterceptor: AnyObject {
+    func intercept(response: ContentLoader.Content) -> ContentLoader.LoadContentPublisher
+}
+
+fileprivate enum ImageCacheKey: CaseIterable {
+    case raw
+    case videoPreview
+
+    static func cacheKey(for url: URL, prefix: ImageCacheKey) -> String {
+        switch prefix {
+        case .raw:
+            return url.absoluteString
+        case .videoPreview:
+            return "video-preview-\(url.absoluteString)"
+        }
+    }
+}
+
+class AvAssetContentLoaderResponseInterceptor: ContentLoaderResponseInterceptor {
+    private let cache: ContentCacheStorage
+
+    var compressedImageSize: CGSize = CGSize(width: 400, height: 400)
+
+    init(cache: ContentCacheStorage) {
+        self.cache = cache
+    }
+
+    func intercept(response: ContentLoader.Content) -> ContentLoader.LoadContentPublisher {
+        switch response {
+        case .svg, .image:
+            return .just(.done(response))
+        case .video(let video):
+            return AVAsset(url: video.url)
```

Comment:
The videos' still preview generation might fail since our input is a URL. Do we need to retry (where we access it, I think in `buildFetchPublisher()`) if we know it's a video but yet have no still preview?